doc/swish/pregexp.tex

```diff
@@ -0,0 +1,817 @@
+
+The \code{yack} in \code{yackety} doesn't end at a word boundary so it
+isn't matched.  The second \code{yack} does and is.
+
+The metasequence \code{{\textbackslash}B} has the opposite effect to
+\code{{\textbackslash}b}.  It asserts that a word boundary does not
+exist.
+
+\code{(pregexp-match-positions "an{\textbackslash\textbackslash}B" "an
+  analysis")} $\Rightarrow$ \code{((3 . 5))}
+
+The \code{an} that doesn't end in a word boundary is matched.
+
+\subsection {Characters and Character Classes}
+
+Typically a character in the regexp matches the same character in the
+text string.  Sometimes it is necessary or convenient to use a regexp
+metasequence to refer to a single character.  Thus, metasequences
+\code{{\textbackslash}n}, \code{{\textbackslash}r},
+\code{{\textbackslash}t}, and \code{{\textbackslash}.}  match the
+newline, return, tab, and period characters respectively.
+
+The \emph{metacharacter} period (\code{.}) matches \emph{any}
+character other than newline.
+
+\code{(pregexp-match "p.t" "pet")} $\Rightarrow$ \code{("pet")}
+
+It also matches \code{pat}, \code{pit}, \code{pot}, \code{put}, and
+\code{p8t} but not \code{peat} or \code{pfffft}.
+
+A \emph{character class} matches any one character from a set of
+characters. A typical format for this is the \emph{bracketed character
+  class} \code{[}\ldots\code{]}, which matches any one character from
+the non-empty sequence of characters enclosed within the
+brackets.\footnote{Requiring a bracketed character class to be
+  non-empty is not a limitation, since an empty character class can be
+  more easily represented by an empty string.}  Thus
+\code{"p[aeiou]t"} matches \code{pat}, \code{pet}, \code{pit},
+\code{pot}, \code{put} and nothing else.
+
+Inside the brackets, a hyphen (\code{-}) between two characters
+specifies the ASCII range between the characters.  For example,
+\code{"ta[b-dgn-p]"} matches \code{tab}, \code{tac}, \code{tad},
+\emph{and} \code{tag}, \emph{and} \code{tan}, \code{tao}, \code{tap}.
+
+An initial caret (\code{\textasciicircum}) after the left bracket
+inverts the set specified by the rest of the contents, i.e., it
+specifies the set of characters \emph{other than} those identified in
+the brackets.  For example, \code{"do[{\textasciicircum}g]"} matches
+all three-character sequences starting with \code{do} except
+\code{dog}.
+
+Note that the metacharacter \code{\textasciicircum} inside brackets
+means something quite different from what it means outside.  Most
+other metacharacters (\code{.}, \code{*}, \code{+}, \code{?}, etc.)
+cease to be metacharacters when inside brackets, although you may
+still escape them for peace of mind.  \code{-} is a metacharacter only
+when it's inside brackets, and neither the first nor the last
+character.
+
+Bracketed character classes cannot contain other bracketed character
+classes (although they contain certain other types of character
+classes-—-see below).  Thus a left bracket (\code{[}) inside a
```

Comment:
The middle hyphen-like character renders some odd characters for me. `classes-âĂŤ-see below`

Reply:
Got it, thanks!